frontend/src/metabase/dashboard/actions/auto-wire-parameters/toasts.ts

```diff
@@ -0,0 +1,88 @@
+import { t } from "ttag";
+import _ from "underscore";
+import type {
+  DashboardCard,
+  DashboardId,
+  DashCardId,
+  ParameterId,
+} from "metabase-types/api";
+import type { Dispatch } from "metabase-types/store";
+import {
+  setDashCardAttributes,
+  setMultipleDashCardAttributes,
+} from "metabase/dashboard/actions";
+// import { disableAutoWireForParameterTarget } from "metabase/dashboard/actions/auto-wire-parameters/actions";
+import { getParameterMappings } from "metabase/dashboard/actions/auto-wire-parameters/utils";
+import { addUndo, dismissUndo } from "metabase/redux/undo";
+
+export const AUTO_WIRE_TOAST_ID = _.uniqueId();
+
+type ShowAutoWireParametersToastType = {
+  dashboardId: DashboardId;
+  parameter_id: ParameterId;
+  sourceDashcardId: DashCardId;
+  modifiedDashcards: DashboardCard[];
+};
+
+export const showAutoWireParametersToast =
+  ({ parameter_id, modifiedDashcards }: ShowAutoWireParametersToastType) =>
+  (dispatch: Dispatch) => {
+    dispatch(
+      addUndo({
+        id: AUTO_WIRE_TOAST_ID,
+        message: t`This filter has been auto-connected with questions with the same field.`,
+        actionLabel: t`Undo auto-connection`,
+        undo: true,
+        action: () => {
+          dispatch(
+            setMultipleDashCardAttributes({
+              dashcards: modifiedDashcards.map(dc => ({
+                id: dc.id,
+                attributes: {
+                  parameter_mappings: getParameterMappings(
```

Comment:
any reason we can't just use dc.parameter_mappings?
```diff
  setMultipleDashCardAttributes({
              dashcards: modifiedDashcards.map(dc => ({
                id: dc.id,
                attributes: {
-		 								parameter_mappings: getParameterMappings(
-                     dc,
-                     parameter_id,
-                     dc.card.id,
-                     null,
-                   ),
+                   parameter_mappings: dc.parameter_mappings,
                },
              })),
            }),
```
This should restore things as they were at the time of the toast being showed right?

Reply:
Nice catch! It does the job 😄